components/aks-mis/demo.tf

```diff
@@ -0,0 +1,37 @@
+# Update -target in azure-pipelines.yaml
+data "azurerm_user_assigned_identity" "sops_demo_mi" {
+  name                = "aks-${var.environment}-mi"
+  resource_group_name = "genesis-rg"
+}
+
+locals {
+  external_dns = {
+    # Resource Groups to add Reader permissions for external dns to
+    resource_groups = toset([
+      "/subscriptions/ed302caf-ec27-4c64-a05e-85731c3ce90e/resourceGroups/reformMgmtRG",
+      "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/core-infra-intsvc-rg"
+    ])
+    # Demo DNS zones to add "DNS Zone Contributor" premissions for external dns to
+    demo = toset([
+      "/subscriptions/ed302caf-ec27-4c64-a05e-85731c3ce90e/resourceGroups/reformMgmtRG/providers/Microsoft.Network/dnszones/demo.platform.hmcts.net",
+      "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/core-infra-intsvc-rg/providers/Microsoft.Network/privateDnsZones/demo.platform.hmcts.net",
+      "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/core-infra-intsvc-rg/providers/Microsoft.Network/privateDnsZones/service.core-compute-demo.internal",
+      "/subscriptions/1baf5470-1c3e-40d3-a6f7-74bfbce4b348/resourceGroups/core-infra-intsvc-rg/providers/Microsoft.Network/privateDnsZones/service.core-compute-idam-demo.internal"
+    ])
```

Comment:
nice, start of refactor to be same as sds?

Reply:
hopefully- makes more sense to do it here now as Tim said - especially given we have more dns zones to manage - will keep the logic of only doing this in the demo environment too this way I think